### PR TITLE
Add admin key to get /network/nodes endpoint

### LINKED ISSUE #8757: HIP-869 Add admin_key to the /network/nodes REST API
### Problem

`admin_key` is not returned in REST API.

### Solution

- Update the `/network/nodes` API to return the `admin_key`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by sdimitrov9**
**Description**:
Adding admin_key to the get /network/nodes REST API

**Related issue(s)**:

Fixes #8757 

**Notes for reviewer**:
Screenshot 2024-11-18 at 11 38 06
Screenshot 2024-11-18 at 11 39 14

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)
Screenshot 2024-11-18 at 11 43 48

### CODE REVIEW

**Comment by SvetBorislavov** [comment]
Hey, the 'admin_key' value structure is not correct. The node's admin_key is of type 'Key', here is the definition inside the protobuf: https://github.com/hashgraph/hedera-protobufs/blob/main/services/state/addressbook/node.proto#L148
The mirror node currently returns this kind of 'Key' structure in the following format:
```
"key": {
        "_type": "ED25519",
        "key": "61f37fc1bbf3ff4453712ee6a305c5c7255955f7889ec3bf30426f1863158ef4"
      }
```
This is an example of the /accounts response:
Here is the whole request: https://testnet.mirrornode.hedera.com/api/v1/accounts?account.id=2159149

The structure should be such because the key may actually be a KeyList with multiple levels

File: hedera-mirror-rest/viewmodel/networkNodeViewModel.js [hunk 1]
```diff
@@ -28,7 +28,8 @@ class NetworkNodeViewModel {
    * @param {NetworkNode} networkNode
    */
   constructor(networkNode) {
-    const {addressBookEntry, nodeStake} = networkNode;
+    const {addressBookEntry, nodeStake, node} = networkNode;
+    this.admin_key = node.adminKey;
```
Comment (steven-sheehy) [inline hedera-mirror-rest/viewmodel/networkNodeViewModel.js:32]:
```suggestion
    this.admin_key = utils.encodeKey(node.adminKey);
```

File: hedera-mirror-rest/model/node.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021-2024 Hedera Hashgraph, LLC
```
Comment (steven-sheehy) [inline hedera-mirror-rest/model/node.js:2]: Copyrights should match the years in which the file was active.

```suggestion
 * Copyright (C) 2024 Hedera Hashgraph, LLC
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/viewmodel/networkNodeViewModel.js (hunk 1 above)
Comment (sdimitrov9) [reply hedera-mirror-rest/viewmodel/networkNodeViewModel.js]: Done.

File: hedera-mirror-rest/model/node.js (hunk 2 above)
Comment (sdimitrov9) [reply hedera-mirror-rest/model/node.js]: Done.

File: hedera-mirror-rest/model/node.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    this.nodeId = node.node_id;
+    this.timestampRange = node.timestamp_range;
+  }
+
+  static tableAlias = 'n';
+  static tableName = 'node';
+
+  static ADMIN_KEY = `admin_key`;
+  static CREATED_TIMESTAMP = `created_timestamp`;
+
```
[the remark this answers is not in the document]
Comment (sdimitrov9) [reply hedera-mirror-rest/model/node.js]: Done.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 4]
```diff
@@ -2862,6 +2881,7 @@ components:
         timestamp:
           $ref: "#/components/schemas/TimestampRange"
       example:
+        admin_key: "421050820e1485acdd59726088e0e4a2130ebbbb70009f640ad95c78dd5a7b38"
```
[the remark this answers is not in the document]
Comment (sdimitrov9) [reply hedera-mirror-rest/api/v1/openapi.yml]: Done.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 5]
```diff
@@ -2774,9 +2774,26 @@ components:
...
+          $ref: "#/components/schemas/TimestampNullable"
+        deleted:
+          description: Whether the node is deleted or not
+          type: boolean
+        node_id:
+          description: An identifier for the node
+          format: int64
+          type: integer
+        timestamp_range:
+          $ref: "#/components/schemas/TimestampRange"
```
[the remark this answers is not in the document]
Comment (sdimitrov9) [reply hedera-mirror-rest/api/v1/openapi.yml]: Done.

File: hedera-mirror-rest/__tests__/viewmodel/networkNodeViewModel.test.js [hunk 6]
```diff
@@ -52,8 +52,13 @@ describe('NetworkNodeViewModel', () => {
       stakeRewarded: 4,
       stakingPeriod: '1654991999999999999',
     },
+    node: {
+      adminKey: null,
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/viewmodel/networkNodeViewModel.test.js:56]: The test should populate a valid key

File: hedera-mirror-rest/__tests__/specs/network/nodes/max-limit.json [hunk 7]
```diff
@@ -367,6 +367,7 @@
   "responseJson": {
     "nodes": [
       {
+        "admin_key": null,
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/specs/network/nodes/max-limit.json:370]: All specs have the admin_key as null. We should populate valid admin keys for at least one file like this one. It should include different variations like ED25519, ECDSA, 3200 immutable key, and one complex key.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/__tests__/viewmodel/networkNodeViewModel.test.js (hunk 6 above)
Comment (sdimitrov9) [reply hedera-mirror-rest/__tests__/viewmodel/networkNodeViewModel.test.js:56]: New tests with valid keys are now added.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/__tests__/viewmodel/networkNodeViewModel.test.js (hunk 6 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/__tests__/viewmodel/networkNodeViewModel.test.js:56]: You added new spec tests, but this test still should have at least one non-null key to specifically test view model.

Comment (sdimitrov9) [reply hedera-mirror-rest/__tests__/viewmodel/networkNodeViewModel.test.js]: Updated.

File: hedera-mirror-rest/__tests__/specs/network/nodes/admin-key-ed25519.json
[the remark this answers is not in the document]
Comment (sdimitrov9) [reply hedera-mirror-rest/__tests__/specs/network/nodes/admin-key-ed25519.json:1]: The unneeded specs are now removed.The existing spec files are updated with the admin_key of the different types as well. The complex type example is a hex value representing a list of keys. It is of type constants.keyTypes.PROTOBUF.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**